language/libra-vm/src/libra_vm.rs

```diff
@@ -459,15 +459,15 @@ pub fn txn_effects_to_writeset_and_events_cached<C: AccessPathCache>(
 
     let ws = WriteSetMut::new(ops)
         .freeze()
-        .map_err(|_| VMStatus::Error(StatusCode::DATA_FORMAT_ERROR))?;
+        .map_err(|_| VMStatus::Error(StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR))?;
```

Comment:
IMO the Move and Libra VM combined should never fail the writeset generation process.

Reply:
same as above